### PR TITLE
feat(MenuItem): allow submenu to open on hover

### PR DESCRIPTION
**Pull request by assuncaocharles**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes https://github.com/microsoft/fluent-ui-react/issues/1783
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

This PR adds `on` property to MenuItem to allow opening the submenu on hover. 

cLG8807FJE

#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/fluentui/react-northstar/src/components/Menu/MenuItem.tsx [hunk 1]
```diff
@@ -228,6 +233,28 @@ export const MenuItem = compose<'a', MenuItemProps, MenuItemStylesProps, {}, {}>
       initialValue: false,
     });
 
+    const getTriggerProps = () => {
+      const triggerProps: any = {};
+      const normalizedOn = _.isArray(on) ? on : [on];
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/Menu/MenuItem.tsx:238]:
```suggestion
```

This is not an array

File: packages/fluentui/react-northstar/src/components/Menu/MenuItem.tsx [hunk 2]
```diff
@@ -228,6 +233,28 @@ export const MenuItem = compose<'a', MenuItemProps, MenuItemStylesProps, {}, {}>
...
+
+      /**
+       * The hover is adding the mouseEnter, mouseLeave events
+       */
+      if (_.includes(normalizedOn, 'hover')) {
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/Menu/MenuItem.tsx:243]:
```suggestion
      if (on === 'hover') {
```

File: packages/fluentui/react-northstar/src/components/Menu/MenuItem.tsx [hunk 3]
```diff
@@ -228,6 +233,28 @@ export const MenuItem = compose<'a', MenuItemProps, MenuItemStylesProps, {}, {}>
       initialValue: false,
     });
 
+    const getTriggerProps = () => {
+      const triggerProps: any = {};
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/Menu/MenuItem.tsx:237]:
```suggestion
      const triggerProps: React.HTMLAttributes<HTMLElement> = {};
```

??

Comment (layershifter) [reply packages/fluentui/react-northstar/src/components/Menu/MenuItem.tsx:237]:
```
const triggerProps: React.HTMLAttributes<HTMLElement> = {
  ...on === 'hover' && {
  onMouseEnter: e => {
    setWhatInputSource(context.target, 'mouse');
    trySetMenuOpen(true, e);
    _.invoke({ onMouseEnter: parentProps.onItemSelect, ...props }, 'onMouseEnter', e, props);
  },
  onMouseLeave: e => {
    trySetMenuOpen(false, e);
    _.invoke(props, 'onMouseLeave', e, props);
  }
  }
}

```

??

Comment (assuncaocharles) [reply packages/fluentui/react-northstar/src/components/Menu/MenuItem.tsx:237]: Fixed

File: packages/fluentui/react-northstar/src/components/Menu/MenuItem.tsx [hunk 4]
```diff
@@ -391,6 +410,8 @@ export const MenuItem = compose<'a', MenuItemProps, MenuItemStylesProps, {}, {}>
       }
     };
 
+    const getRootHandlers = () => (wrapper ? {} : { onClick: handleClick, ...triggerProps });
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/Menu/MenuItem.tsx:413]:
```suggestion
    const rootHandlers = wrapper ? {} : { onClick: handleClick, ...wrapperHandlers };
```

File: packages/fluentui/react-northstar/src/components/Menu/MenuItem.tsx [hunk 5]
```diff
@@ -292,6 +297,20 @@ export const MenuItem = compose<'a', MenuItemProps, MenuItemStylesProps, {}, {}>
     const menuRef = React.useRef<HTMLElement>();
     const itemRef = React.useRef<HTMLElement>();
 
+    const rootHandlers: React.HTMLAttributes<HTMLElement> = {
+      ...(on === 'hover' && {
+        onMouseEnter: e => {
+          setWhatInputSource(context.target, 'mouse');
+          trySetMenuOpen(true, e);
+          _.invoke({ onMouseEnter: parentProps.onItemSelect, ...props }, 'onMouseEnter', e, props);
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/Menu/MenuItem.tsx:305]:
```suggestion
          _.invoke(props, 'onMouseEnter', e, props);
          _.invoke(parentProps, 'onItemSelect', e, props);
```'

Actually we should invoke both

File: packages/fluentui/react-northstar/src/components/Menu/MenuItem.tsx [hunk 6]
```diff
@@ -329,7 +348,7 @@ export const MenuItem = compose<'a', MenuItemProps, MenuItemStylesProps, {}, {}>
       }
       performClick(e);
 
-      _.invoke({ onClick: parentProps.onItemClick, ...props }, 'onClick', e, props);
+      _.invoke({ onClick: parentProps.onItemSelect, ...props }, 'onClick', e, props);
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/Menu/MenuItem.tsx:351]:
```suggestion
          _.invoke(props, 'onClick', e, props);
          _.invoke(parentProps, 'onItemSelect', e, props);
```

File: packages/fluentui/react-northstar/src/components/Menu/MenuItem.tsx [hunk 7]
```diff
@@ -391,6 +410,8 @@ export const MenuItem = compose<'a', MenuItemProps, MenuItemStylesProps, {}, {}>
       }
     };
 
+    const getRootHandlers = () => (wrapper ? {} : { onClick: handleClick, ...rootHandlers });
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/Menu/MenuItem.tsx:413]: Let move this to `rootHandlers` itself

File: packages/fluentui/react-northstar/src/components/Menu/MenuItem.tsx [hunk 8]
```diff
@@ -445,6 +466,28 @@ export const MenuItem = compose<'a', MenuItemProps, MenuItemStylesProps, {}, {}>
...
+      },
+      onClick: (e: React.MouseEvent) => {
+        handleClick(e);
+        _.invoke(predefinedProps, 'onClick', e, props);
+      },
+      ...(on === 'hover' && {
+        onMouseEnter: e => {
+          setWhatInputSource(context.target, 'mouse');
+          trySetMenuOpen(true, e);
+          _.invoke({ onMouseEnter: parentProps.onItemSelect, ...predefinedProps }, 'onMouseEnter', e, props);
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/Menu/MenuItem.tsx:482]:
```suggestion
          _.invoke(predefinedProps, 'onMouseEnter', e, props);
          _.invoke(parentProps, 'onItemSelect', e, props);
```

**Review by layershifter — APPROVED** [review]

**Pull request merged**